src/utils/dates.js

```diff
@@ -30,3 +30,94 @@ export function getDaysBetweenDates(startingDate, endingDate) {
 
 	return daysElapsed;
 }
+
+/**
+ * Sorting compare callback function. Defines sorting order based on purchasing urgency.'
+ * @param {Date} first element up for comparison.
+ * @param {Date} second element up for comparison.
+ */
+function compareItemUrgencyCallback(itemA, itemB) {
+	const today = new Date();
+	const nameOfItemA = itemA.name;
+	const nameOfItemB = itemB.name;
+	const dateNextPurchasedItemA = itemA.dateNextPurchased.toDate();
+	const dateNextPurchasedItemB = itemB.dateNextPurchased.toDate();
+	const daysUntilNextPurchaseItemA = getDaysBetweenDates(
+		today,
+		dateNextPurchasedItemA,
+	);
+	const daysUntilNextPurchaseItemB = getDaysBetweenDates(
+		today,
+		dateNextPurchasedItemB,
+	);
+
+	// sort items by purchasing urgency if urgencies are different
+	if (daysUntilNextPurchaseItemA !== daysUntilNextPurchaseItemB) {
+		return daysUntilNextPurchaseItemA - daysUntilNextPurchaseItemB;
+	}
+	// sort items alphabetically by name if urgencies are the same
+	else {
+		return nameOfItemA.localeCompare(nameOfItemB);
+	}
+}
+
+/**
+ * Filter items into active and inactive categories and sort each category based on purchasing urgency.
+ * @param {Object[]} An array of objects representing the user's unsorted list
+ * @returns {Object[]} An array of objects representing the user's sorted list.
+ */
+export function comparePurchaseUrgency(unsortedList) {
```

Comment:
🔍 A thought: there's some work being repeated in `comparePurchaseUrgency` and `comparePurchaseUrgencyCallback`. If we think about the goal these functions are trying to achieve, there's probably a way that we can avoid repeat work. `comparePurchaseUrgency` runs first, so what can we do in it to prepare the items for sorting, so the sort callback has less to do?

Somewhat related: If we know we want "inactive" items last, we don't have to create a separate array of sorted items and concat the final list back together – we can instruct our sorting algorithm to always send certain items backward (or forward!).

Reply:
Thanks for pointing this out, @mxmason. I reworked these two functions a bit to remove some of the repetitive work and sort "inactive" items last using the sorting algorithm rather than creating a separate array.

Reply:
Solid work @emilysellers! Much more refined. And thanks @mxmason for the nudge.